### ISSUE TITLE
Warp automatically forces me into agent mode when i type certain commands

### ISSUE THREAD

**Issue by harrisrobin** [warpdotdev/Warp#5552]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

When i type certain commands, warp is automatically switching me to agent mode, a feature i never use nor care about. It's frustrating because the UX doesn't give me a clear way to get out of it and i just get stuck in this loop where i have to use agent mode when i don't want to. 

I'm not sure how to disable it and it's not clear how to stop it from behaving this way.

Here's an example:

https://github.com/user-attachments/assets/29a415a4-1628-445b-b442-e9007700bd3d

### To reproduce

1. open warp
2. type in something, like "supabase migrate new.."
3. notice it switches to agent mode. Unsure why, i haven't changed settings or done anything else but i may not understand this feature properly. Eitherway, it seems invasive.

### Expected behavior

For it to not take over and put me in context mode.

### Screenshots

https://github.com/user-attachments/assets/a4f3f30d-8fb7-467b-a6dc-a909f3be379f

### Operating system

MacOS

### Operating system and version

MacOS 15.1.1

### Shell Version

zsh 5.9 (arm64-apple-darwin24.0)

### Current Warp version

v0.2024.12.03.08.02.stable_04

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @harrisrobin,

Sorry about any confusion with the natural language input detection feature of Agent mode. You have some options to workaround/disable it below: 

1. Exiting Agent Mode:
   If you find yourself in Agent Mode and want to exit, you can press `ESC` or `CTRL-C`. On macOS, you can also toggle out of Agent Mode with `CMD-I`, and on Linux, you can use `CTRL-I`.  [Agent Mode Docs](https://docs.warp.dev/features/warp-ai/agent-mode#how-to-enter-agent-mode)

2. Add commands to the denylist:
   If you want to keep auto-detection on for some cases but prevent it from triggering for specific commands, you can add those commands to a denylist. Go to `Settings > AI > Auto-detection denylist` and add the commands that are causing issues. [Agent Mode Docs](https://docs.warp.dev/features/warp-ai/agent-mode#auto-detection-for-natural-language-and-configurable-settings)

3. Disable Auto-detection:
   The easiest way to prevent Warp from automatically switching to Agent Mode is to turn off the auto-detection feature. You can do this by going to `Settings > AI > Input Auto-detection` and disabling it. [Agent Mode Docs](https://docs.warp.dev/features/warp-ai/agent-mode#auto-detection-for-natural-language-and-configurable-settings)

It's worth noting that the auto-detection feature is completely local, and none of your input is sent to AI unless you press `ENTER` in Agent Mode [Warp documentation](https://docs.warp.dev/features/warp-ai/agent-mode).

Closing as this is a known issue and there is a workaround to the issue.

**Comment by zakini** [comment]
Exiting agent mode has been trickier for me lately. I used to be able to CTRL+C to exit and keep my input, but now that clears my input, so I can end up in a loop of trying to type out a command and agent mode getting in the way of me running that

ESC doesn't work for me if I have any input, it only works if I haven't typed anything in

The only thing that works for me is CMD+I, but that's not as intuitive as the other two others

Can anyone at Warp confirm if this is a deliberate change or a bug?

**Comment by Gabbo7474** [comment]
It mostly happens when using some of my aliases, would be interesting if warp could fetch the aliases and not trigger auto-detection when it sees an alias is used.

**Comment by frezbo** [comment]
> It mostly happens when using some of my aliases, would be interesting if warp could fetch the aliases and not trigger auto-detection when it sees an alias is used.

yes this happens with an alias, would be nice to automatically add it to deny list

**Comment by hayata-yamamoto** [comment]
+1

**Comment by dannyneira** [comment]
interesting @jcampbell, can you post the command (please redacy any important info)? 
Are you using Universal or Classic input? 

Also, please check your Settings > AI and make sure Natural Language Detection is disabled. Also provide a screenshot of the redacted command triggering the agent, even with NLD disabled

**Comment by jcampbell** [comment]
I can confirm that the autodetect setting is turned off. The command does not have to be fancy at all, in fact I believe it is purely about the clipboard, and the fact that warp is unable to handle RTF as text (it seems to both paste the text and also try to treat it as an image).

See attached screenshot, which shows the contents of my clipboard, which I have just pasted into the command window, and the fact that they have triggered AI mode. Note from the clipboard that the content type is RTF.

<img width="989" height="479" alt="Image" src="https://github.com/user-attachments/assets/0819add7-0895-4758-ac52-7d40a5b7e0e2" />

**Comment by dannyneira** [comment]
Interesting, I can't reproduce  this by copying from a RTF with Natural language detection enabled or disabled, makes no difference and they are recognized as a shell command in both cases. Make sure you're on the latest version of Warp, and try deleting the natural language settings with `defaults delete dev.warp.Warp-Stable AIAutoDetectionEnabled` then restart Warp, and disable NLD again from Settings > AI menu and try again.

https://github.com/user-attachments/assets/336ed4ab-bf66-4740-a717-48798c950ce3

If the issue persists, please provide a recording of the problem along with [logs](https://docs.warp.dev/support-and-billing/sending-us-feedback#gathering-warp-logs).

**Comment by jcampbell** [comment]
I must have jumped to assuming RTF was enough. Here's a recording of the problem (pasting from excel) including confirmation of being on the latest version and the correct settings. I can provide logs if you still cannot reproduce; let me know how you'd like them delivered.

![Image](https://github.com/user-attachments/assets/17a2a46a-b1eb-44d3-a55b-63c122f350d7)

**Comment by benjamin-smith** [comment]
+1 for disabling by default. Or, if I execute a command that Warp thinks should go to the agent, have some sort of non-invasive popup/tooltip that gives me an opportunity to confirm that is what I want. Especially if the command I was trying to run did not exit cleanly.

**Comment by kargarisaac** [comment]
soooooo annoying. I want to disable this smar (not smart actually) feature!!!!!

**Comment by pro4tlzz** [comment]
This is so annoying, it should be disabled by default!

**Comment by torchsec-roni** [comment]
The feature that I am interested in is AI summary in the sidebar. I am not interested in AI mode. I want to disable "AI Mode" inputs in my terminal and have tried a bunch of different configurations for this. In the meantime, I am stuck between Warp (for the nice AI summary UX in the sidebar while I run Claude Code) and classic Macbook Terminal for actual terminal sessions.
